test/functional/feature_assumeutxo.py

```diff
@@ -0,0 +1,246 @@
+a serialized version of the UTXO set at a certain height, which corresponds
+to a hash that has been compiled into bitcoind.
+
+The assumeutxo value generated and used here is committed to in
+`CRegTestParams::m_assumeutxo_data` in `src/chainparams.cpp`.
+
+## Possible test improvements
+
+- TODO: test submitting a transaction and verifying it appears in mempool
+- TODO: test what happens with -reindex and -reindex-chainstate before the
+      snapshot is validated, and make sure it's deleted successfully.
+
+Interesting test cases could be loading an assumeutxo snapshot file with:
+
+- TODO: An invalid hash
+- TODO: Valid hash but invalid snapshot file (bad coin height or truncated file or
+      bad other serialization)
+- TODO: Valid snapshot file, but referencing an unknown block
+- TODO: Valid snapshot file, but referencing a snapshot block that turns out to be
+      invalid, or has an invalid parent
+- TODO: Valid snapshot file and snapshot block, but the block is not on the
+      most-work chain
+
+Interesting starting states could be loading a snapshot when the current chain tip is:
+
+- TODO: An ancestor of snapshot block
+- TODO: Not an ancestor of the snapshot block but has less work
+- TODO: The snapshot block
+- TODO: A descendant of the snapshot block
+- TODO: Not an ancestor or a descendant of the snapshot block and has more work
+
+"""
+from test_framework.test_framework import BitcoinTestFramework
+from test_framework.util import assert_equal, wait_until_helper
+
+START_HEIGHT = 199
+SNAPSHOT_BASE_HEIGHT = 299
+FINAL_HEIGHT = 399
+COMPLETE_IDX = {'synced': True, 'best_block_height': FINAL_HEIGHT}
+
+
+class AssumeutxoTest(BitcoinTestFramework):
+
+    def set_test_params(self):
+        """Use the pregenerated, deterministic chain up to height 199."""
+        self.num_nodes = 3
+        self.rpc_timeout = 120
+        self.extra_args = [
+            [],
+            ["-fastprune", "-prune=1", "-blockfilterindex=1", "-coinstatsindex=1"],
+            ["-txindex=1", "-blockfilterindex=1", "-coinstatsindex=1"],
+        ]
+
+    def setup_network(self):
+        """Start with the nodes disconnected so that one can generate a snapshot
+        including blocks the other hasn't yet seen."""
+        self.add_nodes(3)
+        self.start_nodes(extra_args=self.extra_args)
+
+    def run_test(self):
+        """
+        Bring up two (disconnected) nodes, mine some new blocks on the first,
+        and generate a UTXO snapshot.
+
+        Load the snapshot into the second, ensure it syncs to tip and completes
+        background validation when connected to the first.
+        """
+        n0 = self.nodes[0]
+        n1 = self.nodes[1]
+        n2 = self.nodes[2]
+
+        # Mock time for a deterministic chain
+        for n in self.nodes:
+            n.setmocktime(n.getblockheader(n.getbestblockhash())['time'])
+
+        self.sync_blocks()
+
+        def no_sync():
+            pass
```

Comment:
In 42cae39356fd20d521aaf99aff1ed85856f3c9f3 "test: add feature_assumeutxo functional test"

nit: `self.no_op` already exists for this.

Reply:
This is addressed in #28562